server/auth.js

```diff
@@ -56,14 +63,36 @@ module.exports = (() => {
     login: (req, res) => strategy.login(req, res),
     // local sign-in
     signin: (req, res, done) => passport.authenticate('local-signin', (err, user) => {
-      if (err) {
-        return res.status(401).json({ status: 'error', message: err.statusText });
-      }
-      if (!user) {
-        return res
-          .status(401)
-          .json({ status: 'error', message: 'Invalid Login' });
+      if (err && err.errors && err.errors[0] && err.errors[0]) {
+        const {
+          detail: errDetail,
+          status: errStatus,
+        } = err.errors[0];
+        let responseMessage = '';
+        const responseStatus = errStatus || 500;
+
+        switch (errDetail) {
+          case 'Database unavailable':
```

Comment:
I might remove this one.. But leaving this to give room to future error handling.

Reply:
yes.
My suggestion is doing a mapping between the response code of the API, and some user friendly error message. like, for example:
api returns 404  -> "user does not exist"
api returns 500 -> "the authentication server is temporarily unavailable, please try again later"

On future iterations you can try to map actual API responses, but this should be good enough for now, and an improvement over the current approach